consensus/istanbul/proxy/enode_certificate.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 The Celo Authors
+// This file is part of the celo library.
+//
+// The celo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The celo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the celo library. If not, see <http://www.gnu.org/licenses/>.
+
+package proxy
+
+import (
+	"encoding/hex"
+
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/consensus/istanbul"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// handleEnodeCertificateMsg is invoked by the proxy
+// It will verify that the message is from a validator that is within the
+// validator connection set and then forward it to the proxied validator.
+func (p *proxyEngine) handleEnodeCertificateMsg(peer consensus.Peer, payload []byte) (bool, error) {
+	logger := p.logger.New("func", "handleEnodeCertificateMsg")
+
+	// Verify that this message is not from the proxied validator
+	if p.proxiedValidator != nil && peer.Node().ID() == p.proxiedValidator.Node().ID() {
+		logger.Warn("Got a enodeCertficate message from the proxied validator. Ignoring it", "from", peer.Node().ID())
+		return false, nil
+	}
+
+	msg := new(istanbul.Message)
+
+	// Verify that this message is created by a legitimate validator before forwarding to the proxied validator (it should
+	// be within the validator connection set).
+	if err := msg.FromPayload(payload, p.backend.VerifyValidatorConnectionSetSignature); err != nil {
+		logger.Error("Got an enodeCertificate message signed by a validator not within the validator connection set.", "err", err)
+		return true, istanbul.ErrUnauthorizedAddress
+	}
+
+	// Need to forward the message to the proxied validator
+	logger.Trace("Forwarding consensus message to proxied validator", "from", peer.Node().ID())
+	if p.proxiedValidator != nil {
+		go p.proxiedValidator.Send(istanbul.EnodeCertificateMsg, payload)
```

Comment:
This was basically copying what the original istanbul code did:  https://github.com/getamis/go-ethereum/blob/c7547381b2ea8999e423970d619835c662176790/consensus/istanbul/backend/backend.go#L155

I am not claiming that what they did is correct though.  I created a utility function so that all `Send`s are done within a goroutine and that the errors will be outputted.  At some point, we should "clean up" all the ways that messages are being sent in the `istanbul` package (yet another future work item).